yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -550,6 +550,9 @@ func (yb *TargetYugabyteDB) importBatchFast(conn *pgx.Conn, batch Batch, args *I
 
 		Lets say the importBatchFastRecover fails with some trasient DB error. In that case,
 		caller(fileTaskImporter.importBatch() function) takes care of retrying with importBatchFastRecover
+
+		The violation error will be because of PK in this code path, not Non-PK table with unique constraint.
+		Even if table has PK + UK on different columns, violation will be on PK for sure, given that the data is existing on source database with same schema
```

Comment:
I didn't understand this. How are we sure that this is only going to be PK error? If a table has both a PK and a UK on different columns, it could fail on the UK only.

Reply:
If it is import data command then we are sure right?
Import data file i know anything is possible.

Reply:
Ideally, yes. But consider the cases: 
- the user already ingested some rows into the table before running voyager (for whatever reason). 
- there is no UK on the source, but during migration, user decided to add a UK on the target side. The UK is actually not appropriate. 

In both cases, with the current check (only for UK), we would silently fail, correct?

Reply:
yeah, i meant we would ignore that error.

Reply:
Handling unique constraint case in a subsequent PR, lets get this one merged.

Reply:
Cool 👍 
Okay to do it in a separate PR obviously, ofc but just to make sure we're on the same page, this is essentially a must-have, not a good-to-have. 

> While in case of Unique Key or non-fast path it should be NonRetryable Error and exit.

did not completely understand this statement. getting a UK violation needs to be treated as a NonRetriableError irrespective of fast-path or normal-path. 


also, pls remove the comment in code before merging. Can be confusing later.

Reply:
yes its a must have.